src/coreclr/jit/emitxarch.cpp

```diff
@@ -2822,13 +2857,45 @@ emitter::code_t emitter::emitExtractEvexPrefix(instruction ins, code_t& code) co
             {
                 case 0x66:
                 {
-                    // None of the existing BMI instructions should be EVEX encoded.
-                    // After APX, BMI instructions can be EVEX encoded with NF feature.
+                    // After APX, BMI instructions can be encoded in EVEX.
                     if (IsBMIInstruction(ins))
                     {
-                        // if BMI instructions reaches this part, then it should be APX-EVEX.
-                        // although the opcode of all the BMI instructions are defined with 0x66,
-                        // but it should not, skip this check.
+                        switch (ins)
+                        {
+                            case INS_rorx:
+                            case INS_pdep:
+                            case INS_mulx:
+// TODO: Unblock when enabled for x86
+#ifdef TARGET_AMD64
```

Comment:
Looks like this will get enabled by https://github.com/dotnet/runtime/pull/111778, after which these ifdefs can be removed. Not sure which will get merged first.